src/app/components/PrettyAddress/index.tsx

```diff
@@ -13,17 +14,20 @@ interface Props {
  * Renders a bech32 address split by 4-character blocks
  */
 export const PrettyAddress = memo((props: Props) => {
-  const addr = props.address
-  let prettyAddress = addr
-
   // Bech32 - 1 is used to split the HRP and the pub key
-  if (addr.match(/^oasis1/)) {
-    const parts = addr.split('1')
+  if (!props.address || props.address === '') {
+    return <></>
+  }
+
+  if (isValidAddress(props.address)) {
+    const parts = props.address.split('1')
 
     const hrp = parts[0]
     const publicKey = parts[1].match(/.{1,4}/g)?.join(' ')
```

Comment:
This looks buggy if address contains "1", e.g. `oasis1qq1vzcvxn1js1unsch1me1xz1kr11vcasv1d1eq1`

Reply:
`'oasis1qq1vzcvxn1js1unsch1me1xz1kr11vcasv1d1eq1'.match(/(oasis1|.{1,4})/g).join(' ')` seems to work

Reply:
(updated)

Reply:
Oh, nevermind, Bech32 doesn't have `1` in data part